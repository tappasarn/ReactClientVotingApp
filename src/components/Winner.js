import React from 'react';

export const Winner = (props) => {
    return (
        <div className="winner">
            Winner is {props.winner}!
        </div>
    );
}

Winner.displayName = 'Winner';