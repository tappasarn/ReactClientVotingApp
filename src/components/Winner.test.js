import React from 'react';
import { shallow } from 'enzyme';

import { Winner } from './Winner';

describe('Winner', () => {
    it('should render winner correctly', () => {
        const wrapper = shallow(<Winner winner='option1'/>);
        expect(wrapper).toMatchSnapshot();
    });
});