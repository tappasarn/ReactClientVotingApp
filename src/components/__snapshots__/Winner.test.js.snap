// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner should render winner correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner
    winner="option1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Winner is ",
        "option1",
        "!",
      ],
      "className": "winner",
    },
    "ref": null,
    "rendered": Array [
      "Winner is ",
      "option1",
      "!",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Winner is ",
          "option1",
          "!",
        ],
        "className": "winner",
      },
      "ref": null,
      "rendered": Array [
        "Winner is ",
        "option1",
        "!",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
